pipeline/matching/generators/similar_schedules.py

```diff
@@ -14,6 +14,48 @@ def __init__(self, min_common: int = 0):
         super().__init__(name=GENERATOR_SIMILAR_SCHEDULES)
 
     def generate(self, inp: MatchingInput) -> Iterator[Match]:
-        # TODO: Implement your generator
-        inp.logger.info("Emma Diamon was here!")
-        yield from []
+        potential_matches: Dict[Tuple[UserId, UserId], Match] = {}
+
+        user_1_i = 0
+        user_2_i = 1
+
+        while user_2_i < len(inp.users):
+            user_1 = inp.users[user_1_i]
+            user_2 = inp.users[user_2_i]
+            for availability1 in user_1.schedule:
+                for availability2 in user_2.schedule:
+                    if availability1 == availability2:
+                        # if we have already saved this match, update the metadata
+                        if (user_1.uid, user_2.uid) in potential_matches:
+                            match = potential_matches[(user_1.uid, user_2.uid)]
+                            match.metadata.matchingAvailability.append(
+                                availability1
+                            )
+                            match.metadata.score += 1
+                            potential_matches[
+                                (user_1.uid, user_2.uid)
+                            ] = match  # updated
+                        # if this is a new match, create match class
+                        else:
+                            potential_matches[(user_1.uid, user_2.uid)] = Match(
+                                users={user_1.uid, user_2.uid},
+                                metadata=MatchMetadata(
+                                    generator="similarSchedulesGenerator",
+                                    score=1,
+                                    matchingAvailability=[availability1],
+                                ),
+                            )
+            if user_2_i == len(inp.users):
+                user_1_i += 1
+                user_2_i = user_1_i + 1
+            else:
+                user_2_i += 1
```

Comment:
There could be a simpler way to loop over all the users and their matching availabilities, but I like what you did here because it seems to be working. Let's make sure it works by adding more test cases that will cause both of these code blocks to run.

In the future, if we are sure the code is correct and has tests, then we could try some ways to make the looping simpler.